### PR TITLE
Replace MultiSelect with new MultiSelectField component

### LINKED ISSUE #3384: Migrate MultiSelect components to MultiSelectFieldV2
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

Related: https://github.com/GCTC-NTGC/gc-digital-talent/issues/3128

## Description

MultiSelectFieldV2 is an extension of SelectFieldV2 (both of which are built fully on react-select components). Most of the logic that builds on react-select components exists in SelectFieldV2, and MultiSelectFieldV2 is a minimal extension of it for better developer experience.

Proposed approach:
- write cypress/jest tests that validate that important Select form submissions work
- replace `<MultiSelect>` with `<MultiSelectFieldV2>`
  - wrap v1 around v2, ensuring types play nice together
  - replace all v1 with v2, making any change required if types need massaging
  - rename MultiSelectFieldV2 to SelectField
  - delete MultiSelect
- fix cypress/jest tests as needed

## Acceptance Criteria
- [x] migrate all instance of MultiSelect to MultiSelectFieldV2
- [ ] confirm migrated forms work as expected
- [x] remove MultiSelect component
- [x] remove V2 suffix from MultiSelectFieldV2

   [image]: https://user-images.githubusercontent.com/305339/166570935-6425a615-82c7-4b2f-9840-5c7d698fa2ab.png
   [image-link]: https://example.com
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

### PR DESCRIPTION
**Pull request by patcon**
Resolves #3384 

WIP (this one is pretty cosmetic)

## To Dos

- [ ] resolve conflicts
- [ ] use `rules={{ required: true }}` for defaults (better a11y now from https://github.com/GCTC-NTGC/gc-digital-talent/pull/3536)

### CODE REVIEW

File: frontend/admin/src/js/components/pool/CreatePool.tsx [hunk 1]
```diff
@@ -244,13 +244,12 @@ export const CreatePoolForm: React.FunctionComponent<CreatePoolFormProps> = ({
                 description:
                   "Placeholder displayed on the pool form classifications field.",
               })}
-              name="classifications"
               options={classificationOptions}
               rules={{
                 required: intl.formatMessage(errorMessages.required),
               }}
```
Comment (patcon) [inline frontend/admin/src/js/components/pool/CreatePool.tsx:250]: Re: https://github.com/GCTC-NTGC/gc-digital-talent/pull/3536

If the above gets merged first, we can do up to 150 search-replaces like this and ditch some boilerplate:

```suggestion
              rules={{ required: true }}
```

**Comment by gobyrne** [comment]
@patcon I'm not used to seeing pull requests in the backlog like this. Just checking to make sure it was deliberate.

**Comment by patcon** [comment]
Yeah, it was, and we haven't prioritized the full migration of MultiSelectField and SelectField to V2. We could close until https://github.com/GCTC-NTGC/gc-digital-talent/issues/3384 is pulled out of safebox and prioritized, then re-animate this

Curious @petertgiles' thoughts on this being brought into sprint, as this long delay was his main concern with not doing this at once, and I might have expected it to be pulled in sooner (and apologies for not advocating for this, just kinda going with the flow on other priorities)

**Comment by petertgiles** [comment]
@patcon No, not urgent enough for this sprint.  We've already got a lot of debt going in this sprint and we have a lot of high urgency bug interruptions to focus on.  As long as the issue is being tracked on the board, I'm fine putting it off for a while.

**Comment by patcon** [comment]
Decided in retro to close this in favour of the parent issue. leaving branch available for now.

**Pull request closed without merging**